### PR TITLE
Create exercise matrix

### PR DESCRIPTION
**Pull request by PaulT89**
## Summary

Generated practice exercise Matrix.

Difficulty = 1.  I altered the tests for this exercise - it used to have strings as input (which the student would then have to parse) but that made the test file look weird.  Instead, I made this an arrays-only practice exercise (with the use of loops).

## Checklist
- [x] If docs where changed run `./bin/configlet generate` to ensure all documents are properly generated.
- [x] CI is green

### CODE REVIEW

File: exercises/practice/matrix/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Instructions
+
+Given a string representing a matrix of numbers, return the rows and columns of
```
Comment (verdammelt) [inline exercises/practice/matrix/.docs/instructions.md:3]: Since this no longer takes strings these instructions will need to be modified.

**Review by verdammelt — COMMENTED** [review]
How bad would this exercise be if it included the parsing of the input strings?

Student would need to write a split function that allowed splitting on newline or space. Then creating a 2D array and populating it. Then writing the row and column functions you already have.

The string splitting is probably the hardest - but by this point likely the student has that function in their repertoire...

**Comment by PaulT89** [comment]
> How bad would this exercise be if it included the parsing of the input strings?
> 
> Student would need to write a split function that allowed splitting on newline or space. Then creating a 2D array and populating it. Then writing the row and column functions you already have.
> 
> The string splitting is probably the hardest - but by this point likely the student has that function in their repertoire...

Yeah, the problem isn't the parsing of strings, the problem is the aesthetics of the test file - inputting a matrix as a string means that the matrix becomes unindented and therefore misaligned from the first newline.  It makes it hard to read and messes up the indentation of the test file.

I suppose I could go back and rewrite the test so that there's a helper function that converts the 2d array into a string as it's being fed into the input.  It would solve the 'ugliness' problem, but on the other hand, it wouldn't necessarily be clear to the student that they're dealing with string input, instead of array input.

Another option would perhaps be to write each line as a separate string as part of a list and then weld the bits together.  For example:

````lisp
(let ((matrix (format nil "~{~A~&~}" '("1 2 3"
                                       "4 5 6"
                                       "7 8 9"))))
  ...
````

Would that be better?

File: exercises/practice/matrix/.docs/instructions.md (hunk 1 above)
Comment (PaulT89) [reply exercises/practice/matrix/.docs/instructions.md:3]: Maybe, maybe not - I have new thoughts.  See conversation.

**Comment by PaulT89** [comment]
Okay, I've completely overhauled the exercise - i.e rewritten the example, rewritten the test file, renamed variables, changed the difficulty, etc.

**Review by verdammelt — APPROVED** [review]
I think i like how it is now. approving. Thanks.

**Pull request merged**